third_problem.rb

```diff
@@ -0,0 +1,68 @@
+class ThirdProblem
+  def initialize(numbers_array)
+    @numbers = numbers_array
+  end
+
+  def hasDuplicates
+    if @numbers.empty? || @numbers.length == 1 then
+      return false
+    end
+    duplicate = false
+
+    for it in 0...@numbers.length
+      currentNumber = @numbers[it]
+
+      if alreadySeen(currentNumber) then
+        next
+      end
+
+      if pointingTowardsItself(currentNumber, it)
+        @numbers[it] = 0
+        next
+      end
+
+      targetIndex = currentNumber - 1
+      currentIndex = it
+
+      while !duplicate
```

Comment:
If you read the problem statement carefully it requires not to use nested loops, you are not doing this.

Reply:
Actually maybe i thought too much
"Do not use 2 nesting for loops since the time complexity would be high".
I thought that the reason to not use 2 loops is because of the time complexity. My solution does not increase in time complexity. It's actually O(n) still. In worst case it's O(2n) which is still O(n).
That's why i considered i respected all the conditions.
If you just don't want to have

Reply:
Indeed, but in can be solved only with one loop.